src/AllocatorBuffer.sol

```diff
@@ -61,17 +74,18 @@ contract AllocatorBuffer {
 
     // --- functions ---
 
-    function approve(
-        address token,
-        address spender,
-        uint256 amount
-    ) external auth {
-        TokenLike(token).approve(spender, amount);
-        emit Approve(token, spender, amount);
+    function approve(address asset, address spender, uint256 amount) external auth {
+        TokenLike(asset).approve(spender, amount);
+        emit Approve(asset, spender, amount);
+    }
+
+    function deposit(bytes32 ilk, address asset, uint256 amount) external {
```

Comment:
Not sure if it's worth it but `ilk` can be immutable.

Reply:
Done!